gcc/tests/bats/test.bats

```diff
@@ -0,0 +1,50 @@
+setup() {
```

Comment:
I think setup runs [before each test](https://github.com/sstephenson/bats/blob/03608115df207/README.md#setup-and-teardown-pre--and-post-test-hooks) so these lines get re-executed multiple times.

Reply:
Yeah,  I wasn't sure where to drop these.  test.sh didn't feel appropriate at the time because I was thinking that we probably want to keep it small and consistent to make it easier to abstract at a later date. `setup` seemed the best place despite it being run multiple times since they're lightweight and it ensure they're set correctly before the tests. I wasn't sure if bats made any guarantees about values set outside of setup.